src/Workspaces/Core/Portable/SemanticModelReuse/AbstractSemanticModelReuseLanguageService.cs

```diff
@@ -53,13 +54,9 @@ internal abstract class AbstractSemanticModelReuseLanguageService<
 
                     try
                     {
-                        throw new InvalidOperationException(
-                            $@"Syntax trees should have been equivalent.
----
-{previousSyntaxTree.GetText(CancellationToken.None)}
----
-{currentSyntaxTree.GetText(CancellationToken.None)}
----");
+                        // Avoid including tree contents in exception message for privacy compliance. Instead, include
+                        // in exception type for dump analysis.
+                        throw new SyntaxTreeException("Syntax trees should have been equivalent.", previousSyntaxTree, currentSyntaxTree);
```

Comment:
Just consider making this exception type a nested type here? That way you can also rename it something like "NonEquivalentTreeException" or something.

Reply:
Good idea, done!